### ISSUE TITLE
write error when saving

### ISSUE THREAD

**Issue by gahisee** [giuspen/cherrytree#473]

**Comment by Loris1123** [comment]
Hat kind of network folder do you use? A smb share? Im having the same problem when I try to save the file on a smb share. 
I mounted it the following way:

    sudo mount.cifs //<IP of NAS>/home NAS -o user=<user>,noperm,uid=1000,gid=1000

**Comment by gahisee** [comment]
Yes, it is a smb folder in an Intel NUC machine running Kubuntu. As stated, by using the xml format, the problem went away. The mount command in /etc/fstab:
//<ip addr><folder>  /mnt/NUC  cifs username=<user>,password=<pass>,uid=<uid>,gid=<gid>,file_mode=0777,dir_mode=0777 0 0

**Comment by casainho** [comment]
I was also getting this issue and I was saving cherrytree database file over a samba share on Linux. I found a solution that seems to work well and that is to use a specific option on samba mount about lock/unlock, as explained here: https://support.shotgunsoftware.com/hc/en-us/community/posts/209491808-Getting-database-is-locked-on-NFS-and-CIFS-setup

Here is my config on /etc/fstab (NOTE the nolock option):

//192.167.2.30/home_gpd_2_pocket_pc  /media/HOME_GPD2_POCKET_PC cifs vers=3.0,username=xxx,password=xxx,nolock,iocharset=utf8,uid=1000 0 0

**Comment by gahisee** [comment]
well, the "nolock" seemed to work, but then, the same write failed error!
strange behavior is that if i save again after the error, it behaves it was successful; at least, the asterisk goes away from the title bar.  but if i load again, the changes are not there.
i will test it on another machine.

**Comment by gahisee** [comment]
here's an update:
the "nolock" option definitely helped on one Kubuntu 19.04 machine but not on a laptop running Kubuntu 18.10.  i do not believe the issue depends on Kubuntu but rather on mysql.
when i make a change on the laptop running CherryTree and as soon as i hit Ctrl-S to save, a xxxx-journal file is created on NAS folder and then the write-failed error box appears. when i exit CherryTree, the -journal file disappears.

on the desktop CherryTree, i do not see this -journal file and no write error. so i'm suspecting the issue is related to mysql but exactly what?  i'll keep digging...

**Comment by gahisee** [comment]
my laptop normally connects to the network drive via wi-fi.  i thought to try the wired connection and CherryTree has no write error!  what? how's that?
tried twice and it is true.  can anyone explain this behavior?

**Comment by Bulwagga** [comment]
I just got this error for the first time today.  I'm using Ubuntu and I've had my .mdb file on a raspberry pi samba share for years.   Not sure what changed.  It's the reason why I'm looking through github.  Did the cable connection turn out to be a permanent solution? I'm just getting back to my home projects after 9 months away... Ubuntu has upgraded, Rasbian has upgraded, and I got a new router.  I'm trying to narrow down my search space.  :)
